Allowed the setting of AccountEmail variable

HenryTK opened the pull request:
# Why

Pingdom has single-user and multi-user accounts. When using a multi-user account you need to supply the account owner email in the `Account-Email` header when calling the API. We received this error message when using this library:
```
Error applying plan:

2 error(s) occurred:

* pingdom_check.ping_example: 403 Forbidden: Not permitted for account type.
* pingdom_check.example: 403 Forbidden: Not permitted for account type.

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

We want to be able to authenticate in multi-user Pingdom accounts. For this we need to pass an `AccountEmail` variable, which is the email address of the account holder.

# What 

Added logic to `config.go` to check if the Account Email variable is set and override the client object using the new constructor if appropriate. 

# How to review

To be reviewed in conjuction with the [pull request on go-pingdom library](https://github.com/russellcardullo/go-pingdom/pull/2)

russellcardullo commented on pingdom/config.go:
```diff
@@ -26,8 +27,15 @@ func (c *Config) Client() (*pingdom.Client, error) {
 		c.APIKey = v
 	}
+	if v := os.Getenv("PINGDOM_ACCOUNT_EMAIL"); v != "" {
+		c.APIKey = v
```
This should be `c.AccountEmail = v`.

russellcardullo commented:
Thanks for the PR.  There are a few issues I found with this when testing:

1. Setting account email via an environment variable incorrectly overrides API Key (see inline comment).
2. When using a non-multi-user account Terraform will prompt the user for an account email:

### Terraform file:
```
variable "pingdom_user" {}
variable "pingdom_password" {}
variable "pingdom_api_key" {}

provider "pingdom" {
    user = "${var.pingdom_user}"
    password = "${var.pingdom_password}"
    api_key = "${var.pingdom_api_key}"
}

resource "pingdom_check" "example" {
    type = "http"
    name = "my http check"
    host = "example.com"
    resolution = 5
}
```

### Gives the following:
```
$ terraform plan
provider.pingdom.account_email
 Enter a value:
```

The second item looks similar to another issue I found: https://github.com/hashicorp/terraform/pull/1456

I did test setting the default account email to `""` instead of `nil` and that does prevent the prompt.

In `pingdom/provider.go`:
```
"account_email": &schema.Schema{
    Type:     schema.TypeString,
    Default:  "",
    Optional: true,
    },
```

Would you mind taking a look to see if those changes will work for your use case?

HenryTK commented:
Thanks Russell. Regarding your first point, oops! All fixed and squashed into the commit. And your second point, good stuff, it does work with our use case. I have added this as a second commit to act as an audit of why that default value is in there. 

I will be looking at the check type next, as our project (part of the UK Government Digital Service) only supports HTTPS and this seems to default to HTTP.

russellcardullo commented:
Looks great!  Merging this now.
